### PR TITLE
Draft docs for using let's encrypt with pomerium and dex

### PR DESCRIPTION
**Pull request by fekete-robert**
https://jira-eng-sjc4.cisco.com/jira/browse/OEYE-174

### CODE REVIEW

**Review by bshifter — APPROVED** [review]

File: docs/authentication/customize-dex.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    ```bash
+    kubectl patch observer one-eye --type merge -p "
+    spec:
+      dex:
+        dexchart:
+        valuesSecret:
+          key: config
+          name: <name-of-dex-secret>"
+    ```
```
Comment (pepov) [inline docs/authentication/customize-dex.md]:
```suggestion
    ```bash
    kubectl patch observer one-eye --type merge -p "
    spec:
      dex:
        dexchart:
          valuesSecret:
            key: config
            name: <name-of-dex-secret>"
    ```
```

File: docs/authentication/letsencrypt/index.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    ```yaml
+    spec:
+      authorization: {}
+      certmanager:
+        certManagerOperatorChart: {}
+        enabled: true
+        namepace: cert-manager
+        issuer: <name-of-the-clusterissuer>
+    ```
```
Comment (pepov) [inline docs/authentication/letsencrypt/index.md]: let's make this consistent with the previous examples
```suggestion
    ```bash
    kubectl patch observer one-eye --type merge -p "
    spec:
      certmanager:
        enabled: true
        namepace: cert-manager
        issuer: <name-of-the-clusterissuer>"
    ```
```

File: docs/authentication/letsencrypt/index.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        kubectl apply -f <clusterissuer.yaml>
+        ```
+
+    1. Check that it was created successfully.
+
+        ```bash
+        kubectl get clusterissuers.cert-manager.io
+        ```
+
+1. Edit the Observer custom resource of One Eye, set the **observer.spec.certmanager.issuer** field to the name of the ClusterIssuer, then apply your changes. On the running One Eye cluster, you can do that by running ```kubectl edit observer```.
```
Comment (pepov) [inline docs/authentication/letsencrypt/index.md]:
```suggestion
1. Update the Observer custom resource of One Eye, set the **observer.spec.certmanager.issuer** field to the name of the ClusterIssuer.
```

File: docs/authentication/letsencrypt/index.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    spec:
+      authorization: {}
+      certmanager:
+        certManagerOperatorChart: {}
+        enabled: true
+        namepace: cert-manager
+        issuer: <name-of-the-clusterissuer>
+    ```
+
+1. Delete the old Pomerium secret.
```
Comment (pepov) [inline docs/authentication/letsencrypt/index.md]:
```suggestion
1. Delete the old Pomerium secret and the certificate as well if you previously configured cert manager with the default issuer.
```

File: docs/authentication/letsencrypt/index.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+    ```bash
+    kubectl delete secret one-eye-pomerium-secret
```
Comment (pepov) [inline docs/authentication/letsencrypt/index.md]:
```suggestion
    kubectl delete certificate one-eye-pomerium-ingress
    kubectl delete secret one-eye-pomerium-ingress
```

File: docs/authentication/letsencrypt/index.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    ```
+
```
Comment (pepov) [inline docs/authentication/letsencrypt/index.md:42]:
```suggestion

1. Rerun the reconciler to apply the changes and create the new certificate resource.

    ```bash
    one-eye observer reconcile
    ```

```

File: docs/authentication/letsencrypt/index.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+1. Wait until cert-manager creates a new secret for Pomerium that uses a Let's Encrypt certificate.
+
+    ```bash
+    kubectl get certificate
```
Comment (pepov) [inline docs/authentication/letsencrypt/index.md]:
```suggestion
    kubectl get certificate one-eye-pomerium-ingress
```

File: docs/authentication/letsencrypt/index.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    ```
+
+    In the output, the READY field of ```one-eye-pomerium-ingress``` should be true.
```
Comment (pepov) [inline docs/authentication/letsencrypt/index.md]:
```suggestion
    In the output, the READY field should be true.
```

File: docs/authentication/letsencrypt/index.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+1. Run the following command:
+
+    ```bash
+    one-eye pomerium install --update
+    ```
+
+    Wait until the reconciling is finished.
```
Comment (pepov) [inline docs/authentication/letsencrypt/index.md]:
```suggestion
```

**Review by pepov — CHANGES_REQUESTED** [review]

**Pull request merged**